### PR TITLE
[Asset Inventory] [AWS] Add setup to support integration testing and automatic provisioning for Asset Inventory

### PR DESCRIPTION
**Pull request by kubasobon**
This PR achieves the following goals:
- extends current integration test setup to support Asset Inventory
- adds a test case for a single asset subtype to prove the extension works (AWS EC2s)
- extends _Create Environment_ workflow so that new test environments come with Asset Inventory beta pre-installed
- extends _Tests_ workflow so that Asset Inventory IT test cases run along other IT cases

### Related Issues

Closes https://github.com/elastic/security-team/issues/10322

### CODE REVIEW

**Comment by kubasobon** [comment]
Successful _Create environment_ test run: https://github.com/elastic/cloudbeat/actions/runs/10704804316
Successful _Tests_ run: https://github.com/elastic/cloudbeat/actions/runs/10716528805/job/29715492619 (it "failed", because it is running 8.15 cloudbeat, not using the branch to build it)

File: .github/actions/aws-ci/action.yml
Comment (romulets) [inline .github/actions/aws-ci/action.yml:21]:
```suggestion
    description: 'Keyword expression to use as a pytest -k flag value'
```

Maybe add what values are expected? `aws` and the one specific to asset inventory?

File: .github/actions/aws-ci/action.yml
Comment (romulets) [inline .github/actions/aws-ci/action.yml:17]: If I understand correctly you are re-using the cspm action to asset inventory.

While I like it from a code re-usability stand point, I believe it could get confusing since vulnerability has its own action (correct?).

My point is, if previously we had a config file for each flavour, I'd rather add a new file to the new flavour, or merge them all in one. 

I do see that my suggestion adds more code and maintainability decreases. So I'm on the fence here. Anyway I'm placing the comment to give food for thought

File: .github/actions/aws-ci/action.yml
Comment (romulets) [inline .github/actions/aws-ci/action.yml:44]: Can't we generate `cloudbeat-config` based on the `python-keyword-expression`?

I mean, the default is aws. I could just have `deploy/aws/cloudbeat-${{ inputs.pytest-keyword-expression }}.yml`. This decreases on config. We just need to align the pytest key to the config file name

Comment (romulets) [reply .github/actions/aws-ci/action.yml:44]: I so see further down that the key and file have different names. I'm also ok with you just ignoring the suggestion.

I just thought that one setting only could make it easier to use

File: deploy/aws-asset-inventory/cloudbeat-aws-asset-inventory.yml
Comment (romulets) [inline deploy/aws-asset-inventory/cloudbeat-aws-asset-inventory.yml:51]: Ideally we also have this processor:

```
  - drop_fields:
      fields: ["host.name"]
 ```
 
 This prevents overwritten the host once we add ec2 info for example

**Review by romulets — COMMENTED** [review]
Left some comments. I do think @gurevichdmitry should also have a look at it

File: .github/actions/aws-ci/action.yml
Comment (gurevichdmitry) [inline .github/actions/aws-ci/action.yml:23]: It would be better to update to one of the existing options: `default: cspm_aws` or `default: asset_inventory_aws`. Alternatively, this will not be relevant if new actions are created.

File: .github/actions/aws-ci/action.yml
Comment (gurevichdmitry) [inline .github/actions/aws-ci/action.yml:47]:
```suggestion
        ./cloudbeat -c "$CLOUDBEAT_CONFIG" -d '*' &
```

File: .github/actions/aws-ci/action.yml
Comment (gurevichdmitry) [inline .github/actions/aws-ci/action.yml:59]:
```suggestion
      run: poetry run pytest -k "$KEYWORD" --alluredir=./allure/results/ --clean-alluredir
```

File: .github/workflows/test-environment.yml
Comment (gurevichdmitry) [inline .github/workflows/test-environment.yml:490]:
```suggestion
          scriptname="aws-asset-inventory-linux.sh"
          src="../../$INTEGRATIONS_SETUP_DIR/$scriptname"
          cmd="chmod +x $scriptname && ./$scriptname"
          ../../.ci/scripts/remote_setup.sh -k "$EC2_ASSET_INV_KEY" -s "$src" -h "$ASSET_INV_PUBLIC_IP" -d "~/$scriptname" -c "$cmd"
```

File: .github/workflows/test-environment.yml
Comment (gurevichdmitry) [inline .github/workflows/test-environment.yml:501]:
```suggestion
          aws s3 cp "./aws-asset-inventory-linux.sh" "$S3_BUCKET/aws-asset-inventory-linux.sh"
```

File: tests/integrations_setup/install_aws_asset_inventory_integration.py
Comment (gurevichdmitry) [inline tests/integrations_setup/install_aws_asset_inventory_integration.py:54]: `get_package_version` by default returns the version of the `cloud_security_posture` integration. I believe that asset inventory uses a different package name: `cloud_asset_inventory`.

File: tests/product/tests/data/aws_asset_inventory/aws_ec2_test_cases.py
Comment (gurevichdmitry) [inline tests/product/tests/data/aws_asset_inventory/aws_ec2_test_cases.py:12]: I’m not sure I understand the purpose of the provided test case and its subsequent use in `test_aws_ec2_asset_inventory.py`. Could you please provide an additional test case that uses the same def `test_aws_ec2_asset_inventory` function? In the current test function, it seems like you are searching for all resources using a specific query that is `hard-coded` in the test body.

I believe the intent should be to define an identifier for the resource in the test data, and then use this identifier to search for the resource. Once the resource is found, you can perform assertions on it. This way, you can reuse the same function, with the only change being the resource identifier you search for.

File: .github/actions/aws-ci/action.yml
Comment (kubasobon) [reply .github/actions/aws-ci/action.yml:17]: Let's go with a separate action for asset inventory. This will keep it more consistent and resolve the issue with parameterizing everything.

File: .github/actions/aws-ci/action.yml
Comment (kubasobon) [reply .github/actions/aws-ci/action.yml:44]: Resolved by having a separate action for asset inventory.

File: .github/actions/aws-ci/action.yml
Comment (kubasobon) [reply .github/actions/aws-ci/action.yml:23]: Resolved by having a separate action for asset inventory.

File: .github/actions/aws-ci/action.yml
Comment (kubasobon) [reply .github/actions/aws-ci/action.yml:47]: Resolved by having a separate action for asset inventory.

File: .github/actions/aws-ci/action.yml
Comment (kubasobon) [reply .github/actions/aws-ci/action.yml:59]: Resolved by having a separate action for asset inventory.

File: .github/actions/aws-ci/action.yml
Comment (kubasobon) [reply .github/actions/aws-ci/action.yml:21]: Resolved by having a separate action for asset inventory.

File: .github/workflows/test-environment.yml
Comment (kubasobon) [reply .github/workflows/test-environment.yml:490]: Should I fix it for the entire file, since it's a direct copy of lines above?

File: .github/workflows/test-environment.yml
Comment (kubasobon) [reply .github/workflows/test-environment.yml:501]: Should I fix it for the entire file, since it's a direct copy of lines above?

File: tests/integrations_setup/install_aws_asset_inventory_integration.py
Comment (kubasobon) [reply tests/integrations_setup/install_aws_asset_inventory_integration.py:54]: Nice nice, good catch.

File: tests/product/tests/data/aws_asset_inventory/aws_ec2_test_cases.py
Comment (kubasobon) [reply tests/product/tests/data/aws_asset_inventory/aws_ec2_test_cases.py:12]: I wanted to mirror the definition structure of other tests as close as possible. It is true that there is no need for parameters in asset inventory cases. 
Should I just simplify the structure for asset inventory and get rid of the unnecessary definitions?

File: .github/workflows/test-environment.yml
Comment (gurevichdmitry) [reply .github/workflows/test-environment.yml:490]: This should already be fixed throughout the entire file, but if there are any additional occurrences, please apply the fix to those as well

File: .github/workflows/test-environment.yml
Comment (gurevichdmitry) [reply .github/workflows/test-environment.yml:501]: I'm not certain about the entire file, but at least in this action, yes, please apply the fix. In general, ${{ env. }} should be replaced wherever we are using inline scripting in the actions.

File: tests/product/tests/data/aws_asset_inventory/aws_ec2_test_cases.py
Comment (gurevichdmitry) [reply tests/product/tests/data/aws_asset_inventory/aws_ec2_test_cases.py:12]: Let's discuss this further offline to sync up.

**Comment by gurevichdmitry** [comment]
> Successful _Tests_ run: [link](https://github.com/elastic/cloudbeat/actions/runs/10716528805/job/29715492619) It's red, because it is running 8.15 cloudbeat, not using the branch to build it. The tests were running and checking stuff though.

For testing purposes, you can update the event trigger from `pull_request_target` to `pull_request`, which will ensure that the workflow execution includes your cloudbeat changes.

File: .github/workflows/ci.yml
Comment (gurevichdmitry) [inline .github/workflows/ci.yml:100]: The `aws-account-type` parameter is not required since `single-account` is the default option when not specified.

File: tests/integrations_setup/install_aws_asset_inventory_integration.py
Comment (gurevichdmitry) [inline tests/integrations_setup/install_aws_asset_inventory_integration.py:54]: It seems like there is a formatting issue

Comment (kubasobon) [reply tests/integrations_setup/install_aws_asset_inventory_integration.py:54]: Hmm, how do you mean? It all gets a once-over with black, pylint etc.

File: .github/workflows/ci.yml
Comment (kubasobon) [reply .github/workflows/ci.yml:100]: Oh, right. Dropping that.

File: tests/integrations_setup/install_aws_asset_inventory_integration.py
Comment (gurevichdmitry) [reply tests/integrations_setup/install_aws_asset_inventory_integration.py:54]: nvm, I’ve checked locally, and everything looks fine

**Review by gurevichdmitry — APPROVED** [review]
Great job, @kubasobon, LGTM! After merging the code, please run the `create environment` workflow on the main branch to ensure everything is functioning correctly. Don't forget to manually verify that the integration for asset inventory is installed properly.

**Pull request merged**
